### PR TITLE
Fixed leading zeroes Hex quantity not accepted

### PR DESCRIPTION
**Pull request by NickSneo**
### What does this PR do?
Allow parsing of padded numerical strings with leading zeroes

### Where should the reviewer start?
from decodeQuantity() function in "utils/src/main/java/org/web3j/utils/Numeric.java"

### Why is it needed?
Fixes issues - https://github.com/web3j/web3j/issues/1752

### CODE REVIEW

File: utils/src/main/java/org/web3j/utils/Numeric.java [hunk 1]
```diff
@@ -47,13 +47,25 @@ public static BigInteger decodeQuantity(String value) {
...
+
         try {
-            return new BigInteger(value.substring(2), 16);
+            return new BigInteger(parsePaddedHexQuantity(value), 16);
         } catch (NumberFormatException e) {
             throw new MessageDecodingException("Negative ", e);
         }
     }
 
+    private static String parsePaddedHexQuantity(String value) {
```
Comment (andrii-kl) [inline utils/src/main/java/org/web3j/utils/Numeric.java:58]: I suggest to make simpler solution 
```
public static BigInteger parsePaddedNumberHex(String value) {
        String numWithoutLeadingZeros = cleanHexPrefix(value).replaceFirst("^0+", "");
        return numWithoutLeadingZeros.isEmpty() ? BigInteger.ZERO : new BigInteger(numWithoutLeadingZeros, 16);
    }
```

File: utils/src/test/java/org/web3j/utils/NumericTest.java [hunk 2]
```diff
@@ -72,14 +72,8 @@ public void testQuantityDecode() {
...
-        assertThrows(MessageDecodingException.class, () -> Numeric.decodeQuantity("0x0400"));
-        assertThrows(MessageDecodingException.class, () -> Numeric.decodeQuantity("0x001"));
-    }
-
-    @Test
-    public void testQuantityDecodeLeadingZeroException() {
-
-        assertThrows(MessageDecodingException.class, () -> Numeric.decodeQuantity("0x0400"));
+        assertEquals(Numeric.decodeQuantity("0x0400"), (BigInteger.valueOf(1024L)));
+        assertEquals(Numeric.decodeQuantity("0x001"), (BigInteger.valueOf(1L)));
```
Comment (andrii-kl) [inline utils/src/test/java/org/web3j/utils/NumericTest.java:76]: I would handle more corner cases. 
For example: 0x000

File: utils/src/main/java/org/web3j/utils/Numeric.java (hunk 1 above)
Comment (NickSneo) [reply utils/src/main/java/org/web3j/utils/Numeric.java:58]: Yeah seems Regex could have been a better approach here for replacing leading zeros

Comment (NickSneo) [reply utils/src/main/java/org/web3j/utils/Numeric.java:58]: Yeah coded purely in java for simple solution, but can make it more simpler using regex

File: utils/src/test/java/org/web3j/utils/NumericTest.java (hunk 2 above)
Comment (NickSneo) [reply utils/src/test/java/org/web3j/utils/NumericTest.java:76]: yeah added one more corner case "0x00f"

**Review by andrii-kl — APPROVED** [review]
